src/main/scala/edu/twitter/model/impl/neuralnetwork/NeuralNetworkModel.scala

```diff
@@ -0,0 +1,70 @@
+package edu.twitter.model.impl.neuralnetwork
+
+import java.util
+
+import edu.twitter.model.api.GenericModel
+import org.deeplearning4j.models.embeddings.wordvectors.WordVectors
+import org.deeplearning4j.nn.multilayer.MultiLayerNetwork
+import org.deeplearning4j.text.tokenization.tokenizer.preprocessor.CommonPreprocessor
+import org.deeplearning4j.text.tokenization.tokenizerfactory.{DefaultTokenizerFactory, TokenizerFactory}
+import org.nd4j.linalg.api.ndarray.INDArray
+import org.nd4j.linalg.factory.Nd4j
+import org.nd4j.linalg.indexing.{INDArrayIndex, NDArrayIndex}
+
+/**
+  * Wrapper for Neural Network Model
+  *
+  * @param model       actual model
+  * @param wordVectors word2vec model used to extract the features
+  */
+class NeuralNetworkModel(model: MultiLayerNetwork,
+                         wordVectors: WordVectors) extends GenericModel {
+
+  val name = "NeuralNetwork"
+  val vectorSize = wordVectors.getWordVector(wordVectors.vocab.wordAtIndex(0)).length
+
+  /**
+    * Classify the given tweet.
+    *
+    * @param tweetText target tweet message for classification.
+    * @return 0 for sad & 1 for happy
+    */
+  override def getLabel(tweetText: String): Double = {
+    val features = loadFeaturesFromString(tweetText)
+    val networkOutput = model.output(features)
+    val timeSeriesLength = networkOutput.size(2)
+    val probabilities = networkOutput.get(NDArrayIndex.point(0), NDArrayIndex.all, NDArrayIndex.point(timeSeriesLength - 1))
+    val happy = probabilities.getDouble(0)
```

Comment:
r u sure happy at idx 0 ?

Reply:
Yep, you can check line 151 in the data iterator.